lib/Runtime/Library/JavascriptArray.inl

```diff
@@ -955,45 +955,45 @@ SECOND_PASS:
             return true;
         }
 
-        SparseArraySegment<T> *toSegment = PrepareSegmentForMemOp<T>(toStartIndex, length);
-        if (toSegment == nullptr)
+        const auto isSegmentValid = [length](Js::SparseArraySegment<T>* segment, uint32 startIndex) {
+            uint32 end, segmentEnd;
+            // Check the segment is long enough
+            return (
+                segment &&
+                !UInt32Math::Add(startIndex, length, &end) &&
+                !UInt32Math::Add(segment->left, segment->length, &segmentEnd) &&
```

Comment:
`PrepareSegmentFromMemOp ` is supposed to change the length to match. Since I make no additional change to the segment length after, it has to already by valid and thus this check is just to make sure PrepareSegmentFromMemOp did its job right.